contributed_definitions/nyaml/NXopt.yaml

```diff
@@ -56,348 +43,675 @@ NXopt(NXobject):
-        \@url:
+            Path to the device, which was calibrated.
+            Example: entry/instrument/DEVICE
+        calibration_method:
+          exists: optional
+          doc: 
+            Describe here the approach or method, used to perform the calibration,
+            by a free text.
+        calibration_accuracy(NXdata):
           exists: optional
           doc: |
-            Website of the software.
-      calibration_status(NX_CHAR):
-        doc: |
-          Was a calibration performed? If yes, when was it done? If the
-          calibration time is provided, it should be specified in
-          ENTRY/INSTRUMENT/calibration/calibration_time.
-        enumeration: [calibration time provided, no calibration, within 1 hour, within 1 day, within 1 week]
-      calibration(NXsubentry):
-        exists: recommended
-        doc: |
-          The calibration data and metadata should be described in a separate NeXus file
-          with the link provided in 'calibration_link'.
+            Provide data about the determined accuracy of the device, this may
+            may be a single value or a dataset like wavelength error vs. wavelength etc.
+        calibration_status(NX_CHAR):
+          exists: recommended
+          doc: |
+            Was a calibration performed? If yes, when was it done? If the
+            calibration time is provided, it should be specified in
+            ENTRY/INSTRUMENT/calibration/calibration_time.
+          enumeration: [calibration time provided, no calibration, within 1 hour, within 1 day, within 1 week]
         calibration_time(NX_DATE_TIME):
           exists: optional
           doc: |
             If calibtration status is 'calibration time provided', specify the
             ISO8601 date when calibration was last performed before this
             measurement. UTC offset should be specified.
-        calibration_data_link:
+        calibration_data_link(NXidentifier):
+          exists: recommended
           doc: |
             Link to the NeXus file containing the calibration data and metadata.
-      (NXbeam_path):
+      # Discussion
+      # ##########
+      # The basic idea was: You have always an incident source and a detector.
+      # Therefore, you always have an incident beam and a "detection beam".
+      # These beams can be assigned clear incident and detection angles.
+      #
+      # This information is always of relevance for ellipsometry,
+      # but for Raman experiments, the state of polarization is as well of relevance
+      # Additonally, this is limited for incident, detection and sample normal
+      # to be in one plane. This does not have to be the case for Raman scattering
+      # experiments. Therefore, this is not general enough and may be moved to
+      # NXellipsometry. 
+      # This information may be as well stored in an incident and detection beam
+
+      # beam properties
+      beam_TYPE(NXbeam):
```

Comment:
Consider removing the underscore (not neccessary though).

Reply:
Does the underscore indicate something specific? I like this for readability.

Reply:
No, it's just if you write `beam_TYPE(NXbeam)`, you enforce `beam_0`, `beam_xray`, and the like. If you write `beamTYPE`, you can also write `beam0` and so on (but also still `beam_0`). Depends on what you like better.